### ISSUE TITLE
Regression with 1.0.0 WRT SSH port usage

### ISSUE THREAD

**Issue by treydock** [voxpupuli/beaker-docker#52]
Seems fix I wrote for #44 was broken in #47 as the behavior I reported and fixed in #44 is back.  The original issue was #43 which is back with 1.0.0. If I downgrade to 0.8.4 things work just fine. This is using Docker on a Mac where the host OS won't have access to the docker network so have to connect to SSH using loopback and exposed port and not container IP address and port 22.

```
  Attempting ssh connection to 172.17.0.7, user: root, opts: {:password=>"root", :port=>22, :forward_agent=>false, :auth_methods=>["password", "publickey", "hostbased", "keyboard-interactive"]}
  Warning: Try 1 -- Host 172.17.0.7 unreachable: Errno::ECONNREFUSED - Connection refused - connect(2) for 172.17.0.7:22
  Warning: Trying again in 3 seconds
  Attempting ssh connection to 172.17.0.7, user: root, opts: {:password=>"root", :port=>22, :forward_agent=>false, :auth_methods=>["password", "publickey", "hostbased", "keyboard-interactive"], :logger=>#<Logger:0x00007ff5c811bf28 @level=4, @progname=nil, @default_formatter=#<Logger::Formatter:0x00007ff5c811bed8 @datetime_format=nil>, @formatter=nil, @logdev=#<Logger::LogDevice:0x00007ff5c811be88 @shift_period_suffix=nil, @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDERR>>, @mon_owner=nil, @mon_count=0, @mon_mutex=#<Thread::Mutex:0x00007ff5c811be10>>>, :password_prompt=>#<Net::SSH::Prompt:0x00007ff5c811bde8>, :user=>"root"}
  Warning: Try 2 -- Host 172.17.0.7 unreachable: Errno::ECONNREFUSED - Connection refused - connect(2) for 172.17.0.7:22
  Warning: Trying again in 5 seconds
  Attempting ssh connection to 172.17.0.7, user: root, opts: {:password=>"root", :port=>22, :forward_agent=>false, :auth_methods=>["password", "publickey", "hostbased", "keyboard-interactive"], :logger=>#<Logger:0x00007ff5c811bf28 @level=4, @progname=nil, @default_formatter=#<Logger::Formatter:0x00007ff5c811bed8 @datetime_format=nil>, @formatter=nil, @logdev=#<Logger::LogDevice:0x00007ff5c811be88 @shift_period_suffix=nil, @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDERR>>, @mon_owner=nil, @mon_count=0, @mon_mutex=#<Thread::Mutex:0x00007ff5c811be10>>>, :password_prompt=>#<Net::SSH::Prompt:0x00007ff5c811bde8>, :user=>"root"}
```

**Comment by ghoneycutt** [comment]
@trevor-vaughan @bastelfreak

**Comment by treydock** [comment]
I think this unit test is one I had to update with fixed logic and was reverted in #47 : https://github.com/voxpupuli/beaker-docker/pull/47/files#diff-697bd4a0a23ba7f22ebedacc9793bb8aeff4750d5cef685b3d9b62c30894faa6L539.

Here is SSH attempt that works with 0.8.4, notice how uses loopback and high number port that is what port 22 exports as:

```
  Attempting ssh connection to 127.0.0.1, user: root, opts: {:password=>"root", :port=>"3607", :forward_agent=>false, :auth_methods=>["password", "publickey", "hostbased", "keyboard-interactive"], :logger=>#<Logger:0x00007fae0420d050 @level=4, @progname=nil, @default_formatter=#<Logger::Formatter:0x00007fae0420d000 @datetime_format=nil>, @formatter=nil, @logdev=#<Logger::LogDevice:0x00007fae0420cf60 @shift_period_suffix=nil, @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDERR>>, @mon_owner=nil, @mon_count=0, @mon_mutex=#<Thread::Mutex:0x00007fae0420cee8>>>, :password_prompt=>#<Net::SSH::Prompt:0x00007fae0420cec0>, :user=>"root"}
```
